x-pack/plugins/uptime/public/components/monitor/monitor_title.tsx

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiTitle } from '@elastic/eui';
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { useMonitorId } from '../../hooks';
+import { monitorStatusSelector } from '../../state/selectors';
+import { EnableMonitorAlert } from '../overview/monitor_list/columns/enable_alert';
+import { Ping } from '../../../common/runtime_types/ping';
+
+const isAutogeneratedId = (id: string) => {
+  const autoGeneratedId = /^auto-(icmp|http|tcp)-OX[A-F0-9]{16}-[a-f0-9]{16}/;
+  return autoGeneratedId.test(id);
+};
+
+// For monitors with no explicit ID, we display the URL instead of the
+// auto-generated ID because it is difficult to derive meaning from a
+// generated id like `auto-http-0X8D6082B94BBE3B8A`.
+// We may deprecate this behavior in the next major release, because
+// the heartbeat config will require an explicit ID.
+const getPageTitle = (monId: string, selectedMonitor: Ping | null) => {
```

Comment:
I'd prefer `monitorId` to `monId`. It seems we are using the full word elsewhere, so I think it'd be helpful to keep it consistent and explicit.